hi-ml-histopathology/src/histopathology/utils/analysis_plot_utils.py

```diff
@@ -267,3 +267,34 @@ def add_training_curves_legend(fig: Figure, include_best_epoch: bool = False) ->
                                      color='k', label="Best epoch (val.)"),)
     fig.legend(handles=legend_handles, **legend_kwargs, loc='lower center',
                bbox_to_anchor=(0.5, -0.1), ncol=len(legend_handles))
+
+
+def plot_confusion_matrices(crossval_dfs: Dict[int, pd.DataFrame], class_names: List[str]) -> Figure:
+    """
+    Plot normalized confusion matrices from HyperDrive child runs.
+    :param crossval_dfs: Dictionary of dataframes with cross-validation indices as keys,
+        as returned by :py:func:`collect_crossval_outputs()`.
+    :param class_names: Names of classes.
+    :return: The created `Figure` object.
+    """
+    import seaborn as sns
+    fig, axs = plt.subplots(1, len(crossval_dfs), figsize=(len(crossval_dfs) * 6, 5))
+    for k, tiles_df in crossval_dfs.items():
+        slides_groupby = tiles_df.groupby(ResultsKey.SLIDE_ID)
+        tile_labels_true = slides_groupby[ResultsKey.TRUE_LABEL]
+        # True slide label is guaranteed unique
+        assert all(len(unique_slide_label) == 1 for unique_slide_label in tile_labels_true.unique())
+        labels_true = tile_labels_true.first()
+
+        tile_labels_pred = slides_groupby[ResultsKey.PRED_LABEL]
```

Comment:
Related to my reply to @fepegar above, can we add a warning for non-unique predicted labels analogous to this? https://github.com/microsoft/hi-ml/blob/3f8d9fe429e270ad900ae8b0aaa4538ed2fc5e90/hi-ml-histopathology/src/histopathology/utils/analysis_plot_utils.py#L182-L187

Reply:
Not sure how you want to group here, as we don't use tile_scores

Reply:
I meant adding a similar warning, but replacing `tile_scores` in this snippet by `tile_labels_pred`.